lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -297,7 +297,7 @@ static bool recursiveType(const StructType *ST, const Type *Ty) {
   return Run(Ty);
 }
 
-SPIRVType *LLVMToSPIRVBase::transType(Type *T) {
+SPIRVType *LLVMToSPIRVBase::transType(Type *T, bool IsGlobalSymbol) {
```

Comment:
`IsGlobalSymbol` should be renamed to something type related. For instance, `IsTypeExtendAllowed`?

`transType` is supposed to translate types, so having `IsGlobalSymbol` parameter is confusing.